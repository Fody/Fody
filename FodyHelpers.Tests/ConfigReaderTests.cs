using System.Xml.Linq;
using Fody;
using Xunit;
using Xunit.Abstractions;

public class ConfigReaderTests :
    XunitApprovalBase
{
    [Fact]
    public void Simple()
    {
        var xElementFalse = XElement.Parse("<Node Name='false'/>");
        Assert.False(xElementFalse.ReadBool("Name", false));
        var xElement0 = XElement.Parse("<Node Name='0'/>");
        Assert.False(xElement0.ReadBool("Name", false));
        var xElementTrue = XElement.Parse("<Node Name='true'/>");
        Assert.True(xElementTrue.ReadBool("Name", false));
        var xElement1 = XElement.Parse("<Node Name='1'/>");
        Assert.True(xElement1.ReadBool("Name", false));
        var xElementNone = XElement.Parse("<Node />");
        Assert.False(xElementNone.ReadBool("Name", false));
        var xElementDefault = XElement.Parse("<Node/>");
        Assert.True(xElementDefault.ReadBool("Name", true));
    }

    [Fact]
    public void Whitespace()
    {
        var xElementFalse = XElement.Parse("<Node Name=' '/>");
        var exception = Assert.Throws<WeavingException>(() => xElementFalse.ReadBool("Name", false));
        ApprovalTests.Approvals.Verify(exception.Message);
    }

    [Fact]
    public void Empty()
    {
        var xElementFalse = XElement.Parse("<Node Name=''/>");
        var exception = Assert.Throws<WeavingException>(() => xElementFalse.ReadBool("Name", false));
        ApprovalTests.Approvals.Verify(exception.Message);
    }

    public ConfigReaderTests(ITestOutputHelper output) : 
        base(output)
    {
    }
}