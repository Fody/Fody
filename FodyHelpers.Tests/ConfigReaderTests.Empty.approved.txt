Could not extract 'Name' from xml config. Null, empty, or whitespace are invalid.